src/content/reference/device-os/firmware.md

```diff
@@ -12979,6 +13005,1754 @@ int8_t measurePower() const;
+
+### append() [VariantArray]
+
+{{api name1="Variant::append(val)"}}
+
+Appends an element to the end of a `VariantArray`. This method is implemented in `Variant`. Returns true if the operation succeeded.
+
+```cpp
+// PROTOTYPES
+bool append(Variant val);
+```
+
+### prepend() [VariantArray]
+
+{{api name1="Variant::prepend(val)"}}
+
+Prepend (insert at the beginning) an element to a `VariantArray`. This method is implemented in `Variant`. Returns true if the operation succeeded.
+
+```cpp
+// PROTOTYPES
+bool prepend(Variant val);
+```
+
+
+### insertAt() [VariantArray]
+
+{{api name1="insertAt(index, val)"}}
+
+Inserts an element with value `val` at index `index` (0-based) into a `VariantArray`. This method is implemented in `Variant`. Returns true if the operation succeeded.
+
+```cpp
+// PROTOTYPES
+bool insertAt(int index, Variant val);
+```
+
+
+### removeAt() [VariantArray]
+
+{{api name1="removeAt()"}}
+
+Removes the element at index `index` (0-based) into a `VariantArray`. This method is implemented in `Variant`.
+
+```cpp
+// PROTOTYPES
+void removeAt(int index);
+```
+
+### at() [VariantArray]
+
+{{api name1="at(index)"}}
+
+Returns a copy of the element at index `index` (0-based) into a `VariantArray`. This method is implemented in `Variant`.
+
+```cpp
+// PROTOTYPES
+Variant at(int index) const;
+```
+
+
+## VariantMap
+
+The `VariantMap` class is a [Map](#map) of a [String](#string-class) to a [Variant](#variant). Think of this as a container to store key-value pair of arbitrary data. This is used to hold data before converting it to JSON or CBOR for sending over the network.
+
+```
+// DEFINITION
+typedef Map<String, Variant> VariantMap;
+```
+
+The methods below are implemented in the `Variant` class, so you can use them update a `Variant` that is of type `VariantMap`.
+
+### set() [VariantMap]
+
+{{api name1="Variant::set(key, value)"}}
+
+Add or update an element with key `key` in the map, setting the value to `value`. This method is implemented in `Variant`.
+
+```cpp
+// PROTOTYPES
+bool set(const char* key, Variant val);
+bool set(const String& key, Variant val);
+```
+
+If the stored value of this `Variant` is not a map, it is converted to a map in place prior to the operation.
+
+### remove() [VariantMap]
+
+{{api name1="Variant::remove(key)"}}
+
+Remove an key `key` in the map. Returns true if this variant is a map and the key exists, otherwise returns false. This method is implemented in `Variant`.
+
+```cpp
+// PROTOTYPES
+bool remove(const char* key);
+bool remove(const String& key);
+```
+
+### get() [VariantMap]
+
+{{api name1="Variant::get(key)"}}
+
+Get the value of an element in the map. This method is implemented in `Variant`.
+
+This method copies the value, which is inefficient for complex variant values. operator[] can be used
+to get a reference to the value instead of copying it.
```

Comment:
Let's add a similar note for `at()` that returns the element of the array by value.

Reply:
Fixed